src/unix/darwin.c

```diff
@@ -102,27 +103,32 @@ int uv_exepath(char* buffer, size_t* size) {
 
 
 uint64_t uv_get_free_memory(void) {
-  vm_statistics_data_t info;
+  vm_statistics64_data_t info;
   mach_msg_type_number_t count = sizeof(info) / sizeof(integer_t);
 
-  if (host_statistics(mach_host_self(), HOST_VM_INFO,
-                      (host_info_t)&info, &count) != KERN_SUCCESS) {
+  if (host_statistics64(mach_host_self(), HOST_VM_INFO,
+                      (host_info64_t)&info, &count) != KERN_SUCCESS) {
     return 0;
   }
 
-  return (uint64_t) info.free_count * sysconf(_SC_PAGESIZE);
+  uint64_t used = info.active_count + info.inactive_count + info.compressor_page_count +info.speculative_count + info.wire_count;
+  uint64_t not_used = info.purgeable_count + info.external_page_count;
+  uint64_t available = uv_get_total_memory() - (used - not_used) * sysconf(_SC_PAGESIZE);
```

Comment:
Multiple style issues (long lines, missing blanks, variable declarations should go at the top of the function) and it could use an explanatory comment what those fields mean.

Also, can you fix indentation on lines 110 and 127?

Reply:
Done!